modules/user/client-react/containers/Auth.web.jsx

```diff
@@ -7,19 +7,29 @@ import { withLoadedUser } from './AuthBase';
 const AuthRoute = withLoadedUser(
   ({ currentUser, role, redirect = '/login', redirectOnLoggedIn, component: Component, ...rest }) => {
     const RenderComponent = props => {
+      const redirectBack = props.location.search
```

Comment:
It would be slightly simpler to read if it were:
`const redirectBack = props.location.search || (props.location.pathname ? `?redirectBack=${props.location.pathname}` : '')`

Reply:
@vlasenko Now if you try accessing `/users` and get redirected to `/login?redirectBack=/users`. Then login as `user`, you get redirected back from `/users` do to insufficient access rights to `/`, you are left with `?redirectBack=/users`.

Reply:
Before `redirectBack` was added only when you got redirected on the server, now this gets added also if you click on links you do not have access to.

Reply:
@mitjade I'm not sure I follow. I have tried to reproduce this problem on your PR. I enter url with `/users` at the end as unauthenticated user, then I login and the I get redirected to `/users` just fine. Could you clarify the problem and tell me how to reproduce it with current code in your PR ?

Reply:
Try to login as `user@example.com` not `admin@example.com`.

Reply:
@mitjade The problem is that we don't distinguish between the cases when user is not logged in and when the user is logged in, but role doesn't match.
We have:
```js
      // redirect users unlogged or without sufficient permissions
      return isRoleMatch(role, currentUser) ? (
        <Component currentUser={currentUser} {...props} {...rest} />
      ) : (
        <Redirect to={{ pathname: redirect }} />
      );
```
But we should rather have:
```js
      if (currentUser) {
        return isRoleMatch(role, currentUser) ? (
          <Component currentUser={currentUser} {...props} {...rest} />
        ) : (
          // If user is logged in, but has no permissions to access URL redirect him permanently, without an option to redirect back
          <Redirect to={redirect} />
        );
      } else {
        // If user is not logged in, remember where we should redirect him back
        return <Redirect to={redirect + redirectBack} />;
      }

```